src/tess.py

```diff
@@ -0,0 +1,364 @@
+        """
+        Fetch data from TESS for a given source.
+        Must return a dataframe (or equivalent), even if it is an empty one.
+        This must be implemented by each observatory subclass.
+
+        Parameters
+        ----------
+        cat_row: dict like
+            A row in the catalog for a specific source.
+            In general, this row should contain the following keys:
+            name, ra, dec, mag, filter_name (saying which band the mag is in).
+        kwargs: dict
+            Additional keyword arguments to pass to the fetcher.
+
+        Returns
+        -------
+        data : pandas.DataFrame or other data structure
+            Raw data from the observatory, to be put into a RawData object.
+        altdata: dict
+            Additional data to be stored in the RawData object.
+
+        """
+        # TODO: figure out dstArcSec threshold for flagging close stars
+        DST_THRESHOLD = 10
+        # TODO: figure out mag diff threshold for flagging similar stars
+        MAGDIFF_THRESHOLD = 0.5
+        # TODO: figure out optimal radii for queries
+        
+        # name should be source_id for GAIA catalog entry
+        name = cat_row['name']
+        dec = cat_row['dec']
+        ra = cat_row['ra']
+        mag = cat_row['mag']
+        if mag > 16:
+            # TESS can't see stars fainter than 15 mag
+            print(f"Magnitude of {mag} is too faint for TESS.")
+            return pd.DataFrame()
+
+        cone_radius = kwargs['cone_radius'] \
+                        if 'cone_radius' in kwargs else 0.01
+        sign = ' -' if dec < 0 else ' '
+        coord_str = str(ra) + sign + str(dec)
+        
+        kwargs = {
+            'coordinates': coord_str,
+            'catalog': 'TIC',
+            'radius': cone_radius
+        }
+        catalog_data = self._try_query(Catalogs.query_region, kwargs)
+        if len(catalog_data) == 0:
+            print("No TESS object found for given catalog row.")
+            return pd.DataFrame()
+        
+        # TODO: check for better way of selecting candidate
+        # catalog_data is automatically sorted by dstArcSec, index 0 is closest
+        # also, is GAIAmag guaranteed to exist? 
+        # what if we're not using a GAIA cat_row?
+        candidate = 0
+        while abs(catalog_data['GAIAmag'][candidate] - mag) > MAGDIFF_THRESHOLD:
+            candidate += 1
+        
+        # if there are multiple likely candidates, throw warning
+        close_mags = 0
+        for i in range(len(catalog_data)):
+            gm = catalog_data['GAIAmag'][i]
+            dst = catalog_data['dstArcSec'][i]
+            if abs(gm - mag) < MAGDIFF_THRESHOLD and dst < DST_THRESHOLD:
+                close_mags += 1
+        if close_mags > 1:
+            warnings.warn(f"Multiple ({close_mags}) sources with similar " +
+                            "magnitudes in cone search." +
+                            " Might select incorrect source.")
+
```

Comment:
See comments about the code above. 
You should probably do all of this using numpy arrays instead of loops. 
I thing the warning in case of multiple matches is good. 
I would still choose the one with the closest distance (if you are SURE that they are sorted by distance you can just pick the first one, otherwise sort them yourself).

Reply:
I changed this part into numpy calls, but I feel like the for loop is both faster (albeit not very noticeable) and more readable.

Reply:
I would be very surprised if the for loop is faster than array operations. I'll look at the code again to make sure. 
Either way in this case the speed is probably not important.